### DISCUSSION TITLE
'generated' directory permission is read-only

### DISCUSSION THREAD

**Discussion by MidnightRoller** [magenx/Magento-2-server-installation#178] [Q&A]
Hello,

After running magenx.sh on a fresh minimal Debian 11 and uncomenting SSL lines in the 3 nginx configuration files I am stuck in a 302  https -> http -> https loop on the front-end.

Admin page loads fine prompting to configure 2FA and email sent.

I ran: `php bin/magento deploy:mode:set production`

debug.log contains many errors such as:

> main.CRITICAL: Can't create directory /home/fashion/public_html/generated/code/Magento/CatalogSearch/Model/ResourceModel/Fulltext/Collection/.
Class Magento\CatalogSearch\Model\ResourceModel\Fulltext\Collection\SearchCriteriaResolverFactory generation error: The requested class did not generate properly, because the 'generated' directory permission is read-only. If --- after running the 'bin/magento setup:di:compile' CLI command when the 'generated' directory permission is set to write --- the requested class did not generate properly, then you must add the generated class object to the signature of the related construct method, only.

Prior to this I had the same issue on a Debian 11 minimal install and I made many attempts to correct the issue by changing permissions in generated/, adjusting magento_umask etc but I was unable to resolve the issue and after enabling google 2fa I was just getting a message "Error" after code entry hence the fresh install of OS to see if issue was repeatable.

I would really appreciate it if someone could point me in the right direction to debug this error as I am brand new to Magento.

**Comment by magenx** [comment]
to fix https loop 
make sure your unsecure and secure domain both https
if you still have issues then
you can try to pass HTTPS in php block
`fastcgi_param HTTPS on;`

permissions issue as because you have to create multiple environments and build/deploy from github with actions
do not mess with files on the server

**Reply by MidnightRoller** [reply]
Thank you magenx, I appreciate your help.
